internal/controller/githubissue_controller_test.go

```diff
@@ -0,0 +1,283 @@
+						},
+						{
+							ID:    github.Int64(456),
+							Title: github.String("Issue 2"),
+						},
+					},
+				),
+				mock.WithRequestMatchHandler(
+					mock.PostReposIssuesByOwnerByRepo,
+					http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+						mock.WriteError(
+							w,
+							http.StatusInternalServerError,
+							"github went belly up or something",
+						)
+					}),
+				),
+			)
+
+			ghClient := github.NewClient(MockClient)
+			r := &GithubIssueReconciler{
+				Client:      c,
+				Scheme:      s,
+				Log:         TestLog,
+				IssueClient: &GitHubIssueClient{Client: ghClient},
+			}
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      testIssue.ObjectMeta.Name,
+					Namespace: testIssue.Namespace,
+				},
+			}
+
+			_, err = r.Reconcile(ctx, req)
+			Expect(err).To(HaveOccurred())
+
+			githubIssueReconciled := issuesv1alpha1.GithubIssue{}
+
+			err = c.Get(ctx, req.NamespacedName, &githubIssueReconciled)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(meta.IsStatusConditionTrue(githubIssueReconciled.Status.Conditions, "IssueIsOpen")).To(BeFalse())
+		})
+	})
+})
+
+var _ = Describe("githubIssue controller", func() {
+	Context("When creating githubIssue ", func() {
+		It("Receive error when trying to update an issue", func() {
+			By("editing Issue")
+
+			ctx := context.Background()
+			testIssue := GenerateTestIssue()
+			c, s, err := CreateFakeClient(testIssue)
+			Expect(err).To(BeNil())
+
+			MockClient = mock.NewMockedHTTPClient(
+				mock.WithRequestMatch(
+					mock.GetReposIssuesByOwnerByRepo,
+					[]*github.Issue{
+						{
+							ID:     github.Int64(123),
+							Number: github.Int(123),
+							Title:  github.String(testIssue.Spec.Title),
+						},
+						{
+							ID:     github.Int64(456),
+							Number: github.Int(456),
+							Title:  github.String("Issue 2"),
+						},
+					},
+					[]*github.Issue{
+						{
+							ID:     github.Int64(123),
+							Number: github.Int(123),
+							Title:  github.String(testIssue.Spec.Title),
+						},
+						{
+							ID:     github.Int64(456),
+							Number: github.Int(456),
+							Title:  github.String("Issue 2"),
+						},
+					},
+				),
+				mock.WithRequestMatchHandler(
+					mock.PatchReposIssuesByOwnerByRepoByIssueNumber,
+					http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+						mock.WriteError(
+							w,
+							http.StatusInternalServerError,
+							"github went belly up or something",
+						)
+					}),
+				),
+			)
+
+			ghClient := github.NewClient(MockClient)
+			r := &GithubIssueReconciler{
+				Client:      c,
+				Scheme:      s,
+				Log:         TestLog,
+				IssueClient: &GitHubIssueClient{Client: ghClient},
+			}
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      testIssue.ObjectMeta.Name,
+					Namespace: testIssue.Namespace,
+				},
+			}
+
+			_, err = r.Reconcile(ctx, req)
+			Expect(err).To(HaveOccurred())
```

Comment:
You already have the controller running from what you did in the `suite_test.go`...